tests/compiler/test_typing.py

```diff
@@ -103,31 +103,162 @@ def test_int_type_invalid(snippetcompiler):
     )
 
 
+def test_number_type(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+entity Test:
+    number i = 0
+end
+
+implement Test using std::none
+
+Test(i = 42)
+Test(i = -42)
+Test()
+Test(i = 42.0)
+Test(i = false)
```

Comment:
I don't like this at all, and we'll probably change it soon. But I guess it's good to be alerted if we change it by accident at some point.